app/code/Magento/CatalogInventory/Model/Indexer/ProductPriceIndexFilter.php

```diff
@@ -104,6 +104,14 @@ public function modifyPrice(IndexTableStructure $priceTable, array $entityIds =
             $select->where('stock_item.use_config_manage_stock = 0 AND stock_item.manage_stock = 1');
         }
 
+        if ($entityIds !== null) {
```

Comment:
I think the null check is not needed and can be removed, simplifying code: the `$entityIds` parameter has declared type so in case the `$entityIds` parameter of `modifyPrice()` function is `null`, PHP would throw a fatal error.